lms/djangoapps/grades/new/course_data.py

```diff
@@ -0,0 +1,89 @@
+from lms.djangoapps.course_blocks.api import get_course_blocks
+from xmodule.modulestore.django import modulestore
+from ..transformer import GradesTransformer
+
+
+class CourseData(object):
+    """
+    Houses course data and intelligently gets and caches the
```

Comment:
Follow-up Q to investigate further: How/when is this cache cleared, say on content updates from the course author?